data/src/main/java/com/linkedin/data/schema/JsonBuilder.java

```diff
@@ -282,7 +280,10 @@ public void writeProperties(Map<String, ?> value) throws IOException
   {
     if (value.isEmpty() == false)
     {
-      for (Map.Entry<String, ?> entry : value.entrySet())
+      List<Map.Entry<String, ?>> orderedProperties =  value.entrySet().stream()
```

Comment:
The unit tests seems to imply this, however I'd like to confirm; this method is only for writing custom properties, right? Since this is a generic `JsonBuilder`, does it have any sense of what "properties" are? Or is this just for writing any generic subtree of the JSON object?

Reply:
this is just for "writing  generic subtree of the JSON object", any concerns?

Reply:
Would this affect the order in which all schema attributes are written, even non-custom properties? e.g. [namespace, name, type, fields] vs [type, fields, name, namespace]. Or are those written individually in the schema encoder? I'm concerned that this would affect the ordering of things that aren't custom properties

Reply:
it shouldn't. non-customer properties are written using other code. This code only manage the properties, so all properties written as a trunk. shouldn't affect others